webrender/src/renderer.rs

```diff
@@ -1319,14 +1322,18 @@ impl Renderer {
                          ubo_data: &[T],
                          shader: ProgramId,
                          quads_per_item: usize,
-                         color_texture_id: TextureId,
-                         mask_texture_id: TextureId,
+                         textures: &PrimitiveBatchTextures,
                          max_prim_items: usize,
                          projection: &Matrix4D<f32>) {
         self.device.bind_program(shader, &projection);
         self.device.bind_vao(self.quad_vao_id);
-        self.device.bind_texture(TextureSampler::Color, color_texture_id);
-        self.device.bind_texture(TextureSampler::Mask, mask_texture_id);
+
+        // TODO(nical): This will force all unused slots to be bound to the invalid
+        // TextureId. Should we just leave them out?
```

Comment:
yeah, no need to touch what is not used
if a texture is used and not provided (thus forcing us to bind `invalid()`), that's a bug we should be chasing and fixing